warehouse/slave/worker_job.go

```diff
@@ -96,16 +98,30 @@ func (p *basePayload) sortedColumnMapForAllTables() map[string][]string {
 
 func (p *basePayload) fileManager(config interface{}, useRudderStorage bool) (filemanager.FileManager, error) {
 	storageProvider := warehouseutils.ObjectStorageType(p.DestinationType, config, useRudderStorage)
+	objectStorageConfig := misc.GetObjectStorageConfig(misc.ObjectStorageOptsT{
+		Provider:                    storageProvider,
+		Config:                      config,
+		UseRudderStorage:            useRudderStorage,
+		RudderStoragePrefixOverride: p.RudderStoragePrefix,
+		WorkspaceID:                 p.WorkspaceID,
+	})
+
+	// create a new config map to avoid modifying the original config map
```

Comment:
What is the harm in modifying original config map ?

Reply:
original one is being passed arg from the caller, since we are processing multiple files, modifying the original will result in concurrent map access and results in panic

Reply:
Instead of copying `filemanagerConfig`, can we copy only `config` ?

Reply:
- minor: Copying `config` would lead to copying of lesser fields.
- major: Copying `config` makes the intent more clearer since it is a method argument. `filemanagerConfig` is being created inside the function, so it is bit harder to understand

Reply:
done. cloning config now